src/services/main_service.ts

```diff
@@ -28,12 +28,12 @@ export class MainService extends PluginServiceClass {
         .getCharacteristic(this.hap.Characteristic.RotationSpeed)
         .onGet(() => this.getSpeed())
         .onSet((newState) => this.setSpeed(newState));
+      this.service.setPrimaryService;
     } else {
       this.service = new this.hap.Service.Switch(this.config.name, "Vacuum");
+      this.service.setPrimaryService;
```

Comment:
AFAIK, this is a function, and this line is not calling it. It should be called as `this.service.setPrimaryService(true)`

Reply:
That's what I tried at first, but the checks did fail so it wasn't able to merge that way.